docs: add Vitest documentation (#796)
chromaui/chromatic-docs: src/content/vitest/configure.mdx

| `reporter`            | `boolean \| { verbose: boolean }` | When truthy, use Chromatic's test reporter, which adds information about captured archives to Vitest's output. When `verbose` is `true`, additional details are logged during the test run.<br />Default: `{ verbose: true }` |

## Snapshot options

These options control how Chromatic behaves when capturing snapshots of your components. They can be set at the global, test file, test suite, or test case level (except where noted).

| Option                    | Type                          | Description                                                                                                                                                          |
| ------------------------- | ----------------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------- |

> | `cropToViewport`          | `boolean`                     | When `true`, the snapshot will be clipped to the height of the [viewport](/docs/viewports#playwright).                                                               |

Reviewer: @AriPerkkio, when you're able, can you check and remove this link and others that reference Playwright (including examples) in our docs to avoid confusion? The rationale is that we don't have the full documentation converted to support Vitest, and users may ask about it.

---
title: Configure
slug: vitest/configure
description: Learn about the various configuration options for visual tests for Vitest
sidebar: { order: 3 }
---

import { TabItem, Tabs } from "../../components/Tabs";

# Configure visual tests for Vitest

You can enhance your Vitest and Chromatic tests further by configuring them using the options outlined in the following sections.

> ## Vitest options

Reviewer 2: I think a more general snippet for applying options should go under the first section.

Then this can be renamed to `Disabling auto snapshots`.

The Chromatic Vitest integration supports configuring options globally in your [Vitest configuration file](https://vitest.dev/config/) as follows:

```ts title="vitest.config.ts"
import { defineProject } from "vitest/config";
import { chromaticPlugin } from "@chromatic-com/vitest/plugin";

export default defineProject({
  plugins: [
    chromaticPlugin({
      // 👇 Sets the option at the global level.
      disableAutoSnapshot: true,
    }),
  ],

  // Other configuration options
});
```

> The `disableAutoSnapshot` option can also be configured at the test level as follows:

Reviewer 2: ```suggestion
Snapshots can also be disabled with the `disableAutoSnapshot` utility function as follows:
```

<Tabs>

>   <TabItem label="On test level">

Reviewer 2: ```suggestion
  <TabItem label="Test level">
```

<br />
When called within `test()`, it disables snapshotting for that test:

```jsx title="test/accordion.test.tsx"
import { test } from "vitest";
import { disableAutoSnapshot } from "@chromatic-com/vitest";

test("accordion", async () => {
  // ❌ Not snapshotted
  disableAutoSnapshot(); // Scopes to this test only
  await render(<Accordion />);
});

test("button", async () => {
  // ✅ Snapshotted automatically
  await render(<Button />);
});
```

  </TabItem>

>   <TabItem label="On test suite level">

Reviewer 2: ```suggestion
  <TabItem label="Test suite level">
```

<br />
When called within `describe()`, it disables snapshotting for all tests and suites within that describe block.

```jsx title="test/accordion.test.tsx"
import { describe, test } from "vitest";
import { disableAutoSnapshot } from "@chromatic-com/vitest";

test("accordion", async () => {
  // ✅ Snapshotted automatically
  await render(<Accordion />);
});

describe("button", () => {
  disableAutoSnapshot(); // Scopes to all tests (and nested suites) in this suite

  test("default", async () => {
    // ❌ Not snapshotted
    await render(<Button />);
  });

  describe("as link", () => {
    test("default", async () => {
      // ❌ Not snapshotted
      await render(<Button as={Link} />);
    });
  });
});
```

  </TabItem>

>   <TabItem label="On test file level">

Reviewer 2: ```suggestion
  <TabItem label="Test file level">
```
